doc/source/users_guide/running-a-case.rst

```diff
@@ -457,4 +457,52 @@ long-term archiver tool that supported mass tape storage and HPSS systems.
 However, with the industry migration away from tape archives, it is no longer
 feasible for CIME to support all the possible archival schemes available.
 
+============================
+Data Assimilation and other External Processing
+============================
+
+Users may desire to run a pre or post run task on compute nodes or a data assimilation mechanism.
```

Comment:
This wording feels a bit awkward to me. I suggest breaking it into two sentences:

> CIME provides the ability to run a task on compute nodes before or after a model run. CIME also provides a mechanism to run data assimilation steps as part of a submitted job (i.e., `./case.submit`).